### DISCUSSION TITLE
EMQX nodes not able to start after complete shutdown of cluster

### DISCUSSION THREAD

**Discussion by on-delete** [emqx/emqx#10388] [Q&A] [answered]
Hello, we currently use EMQX in version 4.4.15 and have the following problem:
The EMQX is running as a cluster with 3 nodes in kubernetes. After we scale down the cluster to 0 replicas, and start them back again with 3 replicas, we get the following error on startup:

```
{"time":1681386935161853,"pid":"<0.1654.0>","msg":"ekka_mnesia: still waiting for table(s): [emqx_resource,mqtt_app,emqx_rule,emqx_trace,emqx_banned,mqtt_admin]","level":"warning","gl":"<0.1653.0>"}
{"time":1681386935162643,"pid":"<0.1654.0>","msg":"Check down_nodes should get [] but got ['node2@node2.svc.cluster.local', 'node1@node1.svc.cluster.local']
```

In this scenario this can never be true, as the other nodes are still starting themselves or waiting for the first node to become ready.
We already changed our `podManagementPolicy` from `Parallel` to `OrderedReady`, but it still happens on startup. Only solution is to remove the pvc to get a fresh cluster. How can we solve this problem in case we needed to shutdown the cluster and want to start it again?

**Comment by ieQu1** [comment, marked as answer]
Hello,

Can you check that solution referenced in this ticket works? https://github.com/emqx/emqx/issues/6201
It's crucial that kubernetes registers domain names for the EMQX nodes immediately as they start, without running any readiness probes or anything like that, otherwise the nodes won't be able to reconcile the database state, which is a prerequisite for the successful startup and passing readiness probes.

**Reply by on-delete** [reply]
Hey @ieQu1, thanks for the hint. It looks like this indeed fixed the problem.
